plugins/node/opentelemetry-instrumentation-fastify/src/internal-types.ts

```diff
@@ -15,11 +15,10 @@
  */
 
 import { Span } from '@opentelemetry/api';
-import type { FastifyReply } from 'fastify';
 import { spanRequestSymbol } from './constants';
 
 export type HandlerOriginal = (() => Promise<unknown>) & (() => void);
 
-export type PluginFastifyReply = FastifyReply & {
+export type WithOptionalSpans<T> = T & {
```

Comment:
Can you please explain the motivation for this change?
Is this type exported publicly somewhere?

Reply:
With the previous code, this would add an `import {...} from 'fastify'` in `build/src/internal-types.d.ts`. Any code that imports this file directly or indirectly would get compilation errors.

Now that I think about it, I'm not quite sure if that was a real problem after all. I'm not sure if this file actually gets imported when someone uses the fastify integration.

What's kind-of confusing to me is that `internal-types.ts` and `utils.ts` are mostly private but their type definitions are still produced and made available in the package.

Reply:
This is very confusing. As far as I understand, public API is only what's exported from `index.d.ts`, as this is the file that typescript is importing when a user `import * from '@opentelemetry/instrumentation-fastify`'.
`index.d.ts` can in turn import other files (like `instrumentation.d.ts`) and they need to be transpiled successfully as well (must not contain import from fastify).

This is why we split the types into `types.ts` and `internal-types.ts` - so we can `import type {} from 'fastify'` without this leaking into public API.

> I feel that this new type makes it less likely that someone will expose types from fastify publicly on accident so I believe that it'd be worthwhile to keep it but it's clearly not required. I'll leave this up to you. Would you like me to revert this type?

This is a great goal, but I'm afraid that the codebase is full of these traps, and I don't think it's feasible to clean each and every instrumentation file that we use (those that are not imported from `index.ts`).
So I support preferring readability and simplicity :)
e.g. making the instrumentation functions private so their types are removed from the `instrumentation.d.ts` artifact.

Reply:
I have reverted this type.